homeassistant/components/qbittorrent/helpers.py

```diff
@@ -31,30 +34,33 @@ def format_unix_timestamp(timestamp) -> str:
     return dt_object.isoformat()
 
 
-def format_progress(torrent) -> str:
+def format_progress(torrent: qbittorrentapi.TorrentDictionary) -> str:
     """Format the progress of a torrent."""
-    progress = torrent["progress"]
+    progress: Any = torrent["progress"]
```

Comment:
If I put `progress: float` I got `Type "JsonValueT" is incompatible with type "float"`, this is why I use `Any` here, I don't know if there is another way to do this ?

Reply:
What is the actual type of `torrent["progress"]`? `JsonValueT` implies it can be any valid JSON type. The call to `float` below implies it's a string.

If we are 100% sure `torrent["progress"]` is always a string, do like this:
```
progress = cast(str, torrent["progress"])
```

Please note that Python `cast` does *not* do any kind of actual type conversion, it is *strictly* a hint to the type checker to tell it we are sure we're dealing with a string.

If we are not 100% sure `torrent["progress"]` is always a string, we should instead validate it and raise if it's not a string:

```
progress = torrent["progress"]
if not isinstance(progress, str):
    raise HomeAsisstantError
```

Same for all the other cases called out by @edenhaus

Reply:
Thanks a lot @emontnemery for the explanation !  I'll fix all this.